packages/block-library/src/navigation/edit/index.js

```diff
@@ -220,42 +219,35 @@ function Navigation( {
 	// that automatically saves the menu as an entity when changes are made to the inner blocks.
 	const hasUnsavedBlocks = hasUncontrolledInnerBlocks && ! isEntityAvailable;
 
+	const { navigationFallback } = useSelect( ( select ) => {
```

Comment:
Returning the value of invoking the `getNavigationFallback` function might not be the most ellegant solution here.

Consider whether simply returning a reference to `getNavigationFallback` and then only invoking that within the effect might be less verbose and easy to follow.

Reply:
The question to be asked here is whether we want to react to changes to the fallback or not.

In other words, should the effect trigger again if the fallback id changes? If it's the case, this useSelect should return the id. If it's not, you could just do `const {  getNavigationFallback } = useSelect( coreStore );`  here. 

Also when trying to reply to that question, don't assume that the "fallback id" is fixed. (even if we know that it's fixed now).

Reply:
> ...we want to react to changes to the fallback or not.

💯 

> ...don't assume that the "fallback id" is fixed. (even if we know that it's fixed now).

I believe this answers the question.

Reply:
I've updated this so that we only get the fallback once and it doesn't get called again if that fallback changes during the current editor session.